compiler/aarch64/codegen/ARM64Ops.hpp

```diff
@@ -0,0 +1,1225 @@
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef ARM64OPS_INCL
+#define ARM64OPS_INCL
+
+typedef enum  {
+//		Opcode                                                          	BINARY    	OPCODE    	comments
+/* UNALLOCATED */
+		ARM64Op_bad,                                                    	/* 0x00000000	BAD       	invalid operation */
+/* Branch,exception generation and system Instruction */
+	/* Compare _ Branch (immediate) */
+		ARM64Op_cbz_32_bit,                                             	/* 0x34000000	CBZ       	 */
+		ARM64Op_cbnz_32_bit,                                            	/* 0x35000000	CBNZ      	 */
+		ARM64Op_cbz_64_bit,                                             	/* 0xB4000000	CBZ       	 */
+		ARM64Op_cbnz_64_bit,                                            	/* 0xB5000000	CBNZ      	 */
+	/* Test & branch (immediate) */
+		ARM64Op_tbz,                                                    	/* 0x36000000	TBZ       	 */
+		ARM64Op_tbnz,                                                   	/* 0x37000000	TBNZ      	 */
+	/* Conditional branch (immediate) */
+		ARM64Op_b_cond,                                                 	/* 0x54000000	B_cond    	 */
+	/* Exception generation */
+		ARM64Op_svc,                                                    	/* 0xD4000001	SVC       	 */
+		ARM64Op_hvc,                                                    	/* 0xD4000002	HVC       	 */
+		ARM64Op_smc,                                                    	/* 0xD4000003	SMC       	 */
+		ARM64Op_brk,                                                    	/* 0xD4200000	BRK       	 */
+		ARM64Op_hlt,                                                    	/* 0xD4400000	HLT       	 */
+		ARM64Op_dcps1,                                                  	/* 0xD4A00001	DCPS1     	 */
+		ARM64Op_dcps2,                                                  	/* 0xD4A00002	DCPS2     	 */
+		ARM64Op_dcps3,                                                  	/* 0xD4A00003	DCPS3     	 */
+	/* System */
```

Comment:
These were already included as atomics , If we are gonna remove system specific instructions, we should also prevent direct usage of these too and restrict to atomics so that we don't allow system wide locks but only internal 
I've opened an issue relevant to that earlier https://github.com/eclipse/omr/issues/2539